src/services/montage-helper.service.js

```diff
@@ -5,7 +5,27 @@
 		.module('montage')
 		.factory('montageHelper', montageHelper);
 
-	function montageHelper(montage, authService) {
+	function montageHelper(constants, montage, authService, ngProgressFactory) {
+
+		var pendingRequestCount = 0;
+		var progressBar = ngProgressFactory.createInstance();
+		progressBar.setColor(constants.primaryColor);
+
+		var _request = montage.Client.prototype.request;
+
+		montage.Client.prototype.request = function(...args) {
+			pendingRequestCount++;
+			progressBar.start();
+
+			return _request.bind(this)(...args).then(response => {
+				pendingRequestCount--;
+
+				if (!pendingRequestCount) { progressBar.complete(); }
+
+				return response;
+			});
+		};
+
```

Comment:
Adding all of this logic above the object returned on line 29 contradicts John Papa's [Accessible Members Up Top](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y052) style

> Expose the callable members of the service (its interface) at the top, using a technique derived from the Revealing Module Pattern.

>Why?: Placing the callable members at the top makes it easy to read and helps you instantly identify which members of the service can be called and must be unit tested (and/or mocked).

>Why?: This is especially helpful when the file gets longer as it helps avoid the need to scroll to see what is exposed.

>Why?: Setting functions as you go can be easy, but when those functions are more than 1 line of code they can reduce the readability and cause more scrolling. Defining the callable interface via the returned service moves the implementation details down, keeps the callable interface up top, and makes it easier to read.